source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -288,7 +295,54 @@ void SslSocket::onConnected() { ASSERT(info_->state() == Ssl::SocketState::PreHa
 
 Ssl::ConnectionInfoConstSharedPtr SslSocket::ssl() const { return info_; }
 
+static void keylogCallback(const SSL* ssl, const char* line) {
+  if (ssl == nullptr) {
+    return;
+  }
+  BIO* bio_keylog = static_cast<BIO*> SSL_get_app_data(ssl);
+  BIO_printf(bio_keylog, "%s\n", line);
+  (void)BIO_flush(bio_keylog);
+}
+
+int SslSocket::setSSLKeyLog(bool enable) {
```

Comment:
Can we may separate `enableSslKeyLog()` and `disableSslKeyLog()`.

Reply:
Any advantage to separate it into two functions ?

Reply:
We should clarify the responsibility of the function. But `setSslKeyLog` has dual responsibility to enable/disable key log. This is why I prefer to separate `enableSslKeyLog` and `disableSslKeyLog`